frame/transaction-payment/asset-tx-payment/src/payment.rs

```diff
@@ -140,26 +142,31 @@ where
 	/// Since the predicted fee might have been too high, parts of the fee may be refunded.
 	///
 	/// Note: The `corrected_fee` already includes the `tip`.
+	///
+	/// Returns the fee and tip in the asset used for payment as (fee, tip).
 	fn correct_and_deposit_fee(
 		who: &T::AccountId,
 		_dispatch_info: &DispatchInfoOf<T::RuntimeCall>,
 		_post_info: &PostDispatchInfoOf<T::RuntimeCall>,
 		corrected_fee: Self::Balance,
-		_tip: Self::Balance,
+		tip: Self::Balance,
 		paid: Self::LiquidityInfo,
-	) -> Result<(), TransactionValidityError> {
+	) -> Result<(AssetBalanceOf<T>, AssetBalanceOf<T>), TransactionValidityError> {
 		let min_converted_fee = if corrected_fee.is_zero() { Zero::zero() } else { One::one() };
-		// Convert the corrected fee into the asset used for payment.
+		// Convert the corrected fee and tip into the asset used for payment.
 		let converted_fee = CON::to_asset_balance(corrected_fee, paid.asset())
 			.map_err(|_| -> TransactionValidityError { InvalidTransaction::Payment.into() })?
 			.max(min_converted_fee);
+		let converted_tip = CON::to_asset_balance(tip, paid.asset())
+			.map_err(|_| -> TransactionValidityError { InvalidTransaction::Payment.into() })?;
+
```

Comment:
You're going to need to change the calling code to this - if you read the doc above, it's saying that the tip is already added to the corrected_fee and that's why `_tip` isn't being used in the existing code. With the current PR as is, the event would have the tip lumped in with the fee.

Reply:
The calling code shouldn't need to change. The docs state that the `actual_fee` includes the `tip`, so `inclusion_fee = actual_fee - tip`:
```
/// A transaction fee `actual_fee`, of which `tip` was added to the minimum inclusion fee,
/// has been paid by `who` in an asset `asset_id`.
```

Reply:
@joepetrowski  Yes, in essence this is what I have done in my latest commit.

Reply:
Since `tip <= converted_fee`, there is basically no way this should error though.

Reply:
> @joepetrowski Yes, in essence this is what I have done in my latest commit.

Ah, but this is actually a breaking change on the interpretation of this event! If you see how the native fee payment event is implemented, it returns the `actual_fee` and `tip`, _not_ the `inclusion_fee`: https://github.com/paritytech/substrate/blob/2dbf625/frame/transaction-payment/src/lib.rs#L384L386

I think people should be able to interpret these events using the same logic. Therefore, I would suggest reverting `
06e345e`.

Reply:
Sorry for any confusion. Joe's right.